mbuild/utils/conversion.py

```diff
@@ -1,25 +1,164 @@
 """mBuild conversion utilities."""
+from warnings import warn
+
 import numpy as np
 
 
-def RB_to_OPLS(c0, c1, c2, c3, c4, c5):
-    """Convert Ryckaert-Bellemans type dihedrals to OPLS type.
+def RB_to_OPLS(
+    c0, c1, c2, c3, c4, c5, error_tol=1e-4, value_error_out_of_tol=True
+):
+    r"""Convert Ryckaert-Bellemans type dihedrals to OPLS type.
+
+    .. math::
+    RB_{torsions} &= c_0 + c_1*cos(psi) + c_2*cos(psi)^2 + c_3*cos(psi)^3 + \\
+                  &= c4*cos(psi)^4 + c5*cos(5*psi)^5
+
+    .. math::
+    OPLS_torsions &= \frac{f_0}{2} + \frac{f_1}{2}*(1+cos(t)) + \frac{f_2}{2}*(1-cos(2*t)) + \\
+                  &= \frac{f_3}{2}*(1+cos(3*t)) + \frac{f_4}{2}*(1-cos(4*t))
+
+    where :math:`psi = t - pi = t - 180 degrees`
 
     Parameters
     ----------
     c0, c1, c2, c3, c4, c5 : Ryckaert-Belleman coefficients (in kcal/mol)
+    error_tol : float, default=1e-4
+        The acceptable absolute tolerance between the RB to OPLS conversion.
+        Any value entered is converted to an absolute value.
+    value_error_out_of_tol : bool, default=True
+        This variable determines whether to provide a ValueError if the RB to OPLS
+        conversion is greater than the error_tol term (i.e., value_error_out_of_tol=True),
+        or a warning if the RB to OPLS conversion is greater than the error_tol term
+        (i.e., value_error_out_of_tol=False).
+
+    Returns
+    -------
+    opls_coeffs : np.array, shape=(5,)
+        Array containing the OPLS dihedrals coeffs f0, f1, f2, f3, and f4
+        (in kcal/mol).
+
+
+    Notes
+    -----
+    c5 must equal zero, or this conversion is not possible.
+
+    (c0 + c1 + c2 + c3 + c4 + c5) must equal zero, to have the exact
+    energies represented in the dihedral.
+
+    NOTE: fO IS TYPICALLY NOT IN THE OPLS DIHEDRAL EQUATION AND IS
+    ONLY USED TO TEST IF THIS FUNCTION CAN BE UTILIZED, AND
+    DETERMINE THE fO VALUE FOR LATER ENERGY SCALING IN MOLECULAR DYNAMICS
+    (MD) SIMULATIONS. THIS FUNCTION TESTS IF f0 IS ZERO (f0=0).
+
+    .. warning:: The :math:`\frac{f_{0}}{2}` term is the constant for the OPLS dihedral equation
+        and is only used to test... will be shifted by the :math:`\frac{f_{0}}{2}`
+    If the f0 term is not zero, the dihedral is not an exact conversion;
+    since this constant does not contribute to the force equation,
+    this should provide matching results for MD, but the energy for each
+    dihedral will be shifted by the :math:`\frac{f_{0}}{2}` value.
+    """
+    try:
+        error_tol = abs(float(error_tol))
+    except:
```

Comment:
We should try not to catch the base exception, I think.

Reply:
@justinGilmer   please weight in on this.

Reply:
Or you can do it like you have done below for the bool var

```python3
if not isinstance(error_tol, float):
    raise TypeError

error_tol = abs(error_tol)
```

Reply:
Changed back to how it was mostly was initially,

    if not isinstance(error_tol, float):
        raise TypeError(f"The error_tol variable must be a float, is type {type(error_tol)}.")
    error_tol = abs(error_tol)